src/util.h

```diff
@@ -170,7 +175,7 @@ void DumpBacktrace(FILE* fp);
 #endif
 
 
-#define UNREACHABLE() ABORT()
+#define UNREACHABLE(expr) ASSERT(expr)
```

Comment:
This won't work. The string evaluates to true, meaning the assert won't trigger.

At any rate, I think the message should be prefixed with something like `"Unreachable code reached:"`. I suggest adding a `node::Unreachable()` function that mirrors `node::Assert()`.

Reply:
I have defined the `ASSERT` macro, which just takes the expression and calls `node::Assert` with the right arguments. It seemed to work in my test.
I can try using `node::Unreachable` also.

Reply:
I added `"Unreachable code reached:" prefix for the `UNREACHABLE` macro and also tested again and it works. That's because `expr` is not boolean checked, and so it's okay to pass a string.
I think the confusion here could be naming of the macro `ASSERT`. If that's the case I could rename that. I also thought about creating `node::Unreachable` but don't see how that would be different from `node::Assert` which already exists.

Reply:
If `ASSERT(expr)` doesn't evaluate its expression, doing nothing on truthy, and aborting on falsy, it will surprise every C and C++ developer who sees it.

Reply:
Yeah, that's what I meant by the naming of the `ASSERT` macro. I tried to name it after the `node::Assert` function, which is similar and does not evaluate the expression, but just prints a nice error.
I should certainly change the name for this change, just wondering if the approach feels alright.

Reply:
Updated macro name.